### PR TITLE
This is the April 2022 Update (v2.15.0)

### LINKED ISSUE #722: Add missing DXGI_CREATE_FACTORY_DEBUG
# Summary of feature
Silk.NET.DXGI is doesn't have the constant/flag DXGI_CREATE_FACTORY_DEBUG. 

# Comments
I have to type by hand the 0x1 which is the value of DXGI_CREATE_FACTORY_DEBUG, it would be nice to be able to just grab it from the DXGI class, like DXGI.CpuAccessNone. 

This is useful to make it easier to pass the correct value to DXGI.GetApi().CreateDXGIFactory2
https://docs.microsoft.com/en-us/windows/win32/api/dxgi1_3/nf-dxgi1_3-createdxgifactory2

### LINKED ISSUE #863: How to set Android fullscreen and landscape？
I try set RequestedOrientation on function OnResume, but that's an exception: 
```
System.InvalidOperationException
  Message=Only one SilkActivity may be present throughout the whole application.
```
image

### LINKED ISSUE #867: Unable to load native libraries
Hi everyone,

I was debugging why my application is not able to find `glfwWindowHintString` after updating from silk 1.9 to 2.14

It looks like that `DefaultPathResolver.TryLocateNativeAssetInRuntimesFolder` appends the name of the library twice.

Expected file to load: `bin/Debug/net6.0/runtimes/linux-x64/native/libglfw.so.3`
File it tries to load: `bin/Debug/net6.0/runtimes/linux-x64/native/libglfw.so.3/libglfw.so.3`

I cannot properly debug if this is the root of my problem because moving the file int the directory it expects results in the following build error.
`Microsoft.Common.CurrentVersion.targets(4808, 5): [MSB3024] Could not copy the file "~/.nuget/packages/ultz.native.glfw/3.3.3.1/runtimes/linux-x64/native/libglfw.so.3" to the destination file "bin/Debug/net6.0/runtimes/linux-x64/native/libglfw.so.3", because the destination is a folder instead of a file. To copy the source file into a folder, consider using the DestinationFolder parameter instead of DestinationFiles.`

OS: Ubuntu 18.04
.Net: net6.0

https://github.com/dotnet/Silk.NET/blob/36f3130b2a732b636333f6462704c3882ec0dcdc/src/Core/Silk.NET.Core/Loader/DefaultPathResolver.cs#L203

``` c#
        private bool TryLocateNativeAssetInRuntimesFolder(string name, string baseFolder, out string result)
        {
            static bool Check(string name, string ridFolder, out string result)
            {
                var theoreticalFName = Path.Combine(ridFolder, name);
                if (File.Exists(theoreticalFName))
                {
                    result = theoreticalFName;
                    return true;
                }

                result = null;
                return false;
            }

            foreach (var rid in GetAllRuntimeIds(RuntimeEnvironment.GetRuntimeIdentifier(), DependencyContext.Default))
            {
                if (Check(name, Path.Combine(baseFolder, "runtimes", rid, "native", name), out result))
                {
                    return true;
                }
            }

            result = null;
            return false;
        }
```

### PR DESCRIPTION
**Pull request by Perksey**
- Add DirectStorage bindings (implemented in #855, discussed in #852)
- Add legacy OpenGL 2.0 support for the ImGui extension (implemented in #854)
- Add the ability to customize the functionality/priorities of DefaultPathResolver (implemented here, closes #867)
- Add more auto-generated constants to Assimp, DirectX, and SDL (implemented in #855, properly fixes #722 which was manually fixed in #805 with a bandaid)
- Update to Vulkan 1.3.210 (here)
- Update to latest OpenGL specifications (here)
- Fix Assimp native package being out-of-sync with latest binding (implemented in #850)
- Fix SDL input backend producing different key mappings to the GLFW input backend (implemented here, relates to #737)
- Fix Android Activity restarts not being properly handled by Silk.NET Windowing (implemented here, fixes #863)

### CODE REVIEW

**Review by HurricanKai — COMMENTED** [review]
Honestly amazing work with the path resolver! I've left a handful of comments.

File: build/nuke/Build.PublicApi.cs [hunk 1]
```diff
@@ -16,8 +16,49 @@
...
+    string FormatCmd
+    {
+        get
+        {
+            // hack to use the dotnet-format NOT included with the SDK if it's available. this is useful if the one
+            // that's shipped with the SDK is bugged.
+            var process = OperatingSystem.IsWindows()
+                ? InheritedShell("cmd /c where dotnet-format")
+                : InheritedShell("whereis dotnet-format");
+            process.AssertWaitForExit();
```
[the remark this answers is not in the document]
Comment (Perksey) [reply build/nuke/Build.PublicApi.cs:29]: That's `AssertZeroExitCode`, `AssertWaitForExit` just asserts that the process starts up and exits :)

File: src/Core/Silk.NET.Core/Loader/DefaultPathResolver.cs [hunk 2]
```diff
@@ -24,65 +24,145 @@ namespace Silk.NET.Core.Loader
...
+            MainModuleDirectoryResolver,
+            RuntimesFolderResolver,
+            NativePackageResolver
+        };
 
-        private IEnumerable<string> CoreEnumeratePossibleLibraryLoadTargets(string name, bool skipVersionTraverse = false)
+        /// <summary>
+        /// A list of resolvers to use to obtain paths. 
+        /// </summary>
+        public List<Func<string, IEnumerable<string>>> Resolvers
```
[the remark this answers is not in the document]
Comment (Perksey) [reply src/Core/Silk.NET.Core/Loader/DefaultPathResolver.cs]: Fixed.

File: src/Vulkan/Silk.NET.Vulkan/Structs/DebugUtilsObjectNameInfoEXT.gen.cs [hunk 3]
```diff
@@ -17,7 +17,7 @@
 namespace Silk.NET.Vulkan
 {
     [NativeName("Name", "VkDebugUtilsObjectNameInfoEXT")]
-    public unsafe partial struct DebugUtilsObjectNameInfoEXT : IChainable
+    public unsafe partial struct DebugUtilsObjectNameInfoEXT : IExtendsChain<PipelineShaderStageCreateInfo>
```
[the remark this answers is not in the document]
Comment (Perksey) [reply src/Vulkan/Silk.NET.Vulkan/Structs/DebugUtilsObjectNameInfoEXT.gen.cs:20]: Yes there have been changes upstream in 1.3.210

File: src/Vulkan/Silk.NET.Vulkan/Structs/PipelineLayoutCreateInfo.gen.cs [hunk 4]
```diff
@@ -80,7 +80,7 @@ public PipelineLayoutCreateInfo
         [NativeName("Type", "VkPipelineLayoutCreateFlags")]
         [NativeName("Type.Name", "VkPipelineLayoutCreateFlags")]
         [NativeName("Name", "flags")]
-        public uint Flags;
+        public PipelineLayoutCreateFlags Flags;
```
[the remark this answers is not in the document]
Comment (Perksey) [reply src/Vulkan/Silk.NET.Vulkan/Structs/PipelineLayoutCreateInfo.gen.cs:83]: Yes, they've done a breaking fix upstream.

**Review by HurricanKai — APPROVED** [review]

**Pull request merged**